pkg/k8s/watchers/endpointsynchronizer.go

```diff
@@ -95,11 +95,11 @@ func (epSync *EndpointSynchronizer) RunK8sCiliumEndpointSync(e *endpoint.Endpoin
 					return fmt.Errorf("Kubernetes apiserver is not available")
 				}
 
-				// K8sPodName and K8sNamespace are not always available when an
-				// endpoint is first created, so we collect them here.
-				podName := e.GetK8sPodName()
-				if podName == "" {
-					scopedLog.Debug("Skipping CiliumEndpoint update because it has no k8s pod name")
+				// The CEP name is derived from the K8sPodName and K8sNamespace.
+				// They should always be available if an endpoint belongs to a pod.
+				cepName := e.GetK8sCEPName()
+				if cepName == "" {
+					scopedLog.Debug("Skipping CiliumEndpoint update because it has no k8s cep name")
```

Comment:
Good point, I just checked and we are running this controller for the health and host endpoint which do not have a pod backing it. So emitting a warning or error would cause a lot of unnecessary noise.

We should probably fix that (i.e. not start the controller for such endpoints), but I'll leave that up for future work. I'll open an issue.